File: BE/CleanArchitecture.Domain/Entities/Product.cs
using CleanArchitecture.Domain.Common;

namespace CleanArchitecture.Domain.Entities;

public class Product : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public ProductType Type { get; set; }
    public string Description { get; set; } = string.Empty;
    public string MockupImageUrl { get; set; } = string.Empty;
    public decimal BasePrice { get; set; }
    public bool IsActive { get; set; } = true;

    // Navigation properties
    public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    public virtual ICollection<Mockup> Mockups { get; set; } = new List<Mockup>();
}

public enum ProductType
{
    Shirt = 1,      // Áo
    Hat = 2,        // Mũ
    CanvasBag = 3   // Túi canvas
}

File: FE/src/types/product.ts
export enum ProductType {
  Shirt = 1,
  Hat = 2,
  CanvasBag = 3
}

export enum ProductMode {
  COMBO = 'combo',      // Áo + Decal (149k)
  DECAL_ONLY = 'decal'  // Decal riêng (theo size)
}

export interface Product {
  id: number;
  name: string;
  type: ProductType;
  typeName: string;
  description: string;
  mockupImageUrl: string;
  basePrice: number;
  isActive: boolean;
  createdAt: string;
  mockups: Mockup[];
}

export interface Mockup {
  id: number;
  productId: number;
  name: string;
  imageUrl: string;
  overlayCoordinates: string;
  maxWidth: number;
  maxHeight: number;
  isDefault: boolean;
  sortOrder: number;
}

export interface MockupCoordinates {
  x: number;
  y: number;
  width: number;
  height: number;
}

export interface PriceCalculation {
  productId: number;
  width: number;
  height: number;
  price: number;
}
